include/conf/load.hpp

```diff
@@ -37,17 +38,50 @@ namespace color_coded
       }
     }
 
-    inline args_t load(std::string const &file, std::string const &filetype)
+    inline args_t load_compilation_database(std::string const &file, std::string filename)
     {
-      if(file.empty())
-      { return defaults(filetype); }
+      std::string str;
+      auto database_ptr = ::clang::tooling::JSONCompilationDatabase::loadFromFile(file, str);
+      if(!database_ptr)
+      { return {}; }
+
+      std::vector<fs::path> files = {filename};
+      auto const ext = fs::path(filename).extension();
+      if(ext == ".h" || ext == ".hpp")
+      {
+        files.emplace_back(fs::path(filename).replace_extension("c"));
+        files.emplace_back(fs::path(filename).replace_extension("cpp"));
+      }
+
+      std::vector<::clang::tooling::CompileCommand> compile_commands;
+      for(auto const &file : files)
+      {
+        compile_commands = database_ptr->getCompileCommands(file.string());
+
+        if(!compile_commands.empty())
+        {
+          filename = file.string();
+          break;
+        }
```

Comment:
`files` containes the file and the .cpp and .c files if its a header file. This avoids the ugly nested ifs I had before. If we dont find any flags for the header file, we also check for possible flags in the .cpp or .c file.